### ISSUE TITLE
[Enhancement] Downloading specific tracks of albums

### ISSUE THREAD

**Issue by gisturbance** [DJDoubleD/QobuzDownloaderX-MOD#19]
# **🚀 Enhancement Request**

## **Is your enhancement request related to a problem? Please describe.**
Sometimes when searching for an album, you only want to download specific songs.
Maybe you forgot a single songs name, have a slow or data-limited connection or just don't want all of them.

---

## **Describe the solution you'd like**
Clicking on an album would reveal a list of included tracks (with usual metadata), additionally there's buttons to download each single track and checkboxes to select multiple for a combined download.

---

## **Describe alternatives you've considered**
Search for each song individually or download the whole album and delete unwanted songs.

**Comment by orbittwz** [comment]
you have the option to search for specific track you want...
![image](https://github.com/DJDoubleD/QobuzDownloaderX-MOD/assets/45605734/00d2707a-fbf5-4cc7-ab12-b6f0431048c8)
pretty sure this wasn't available when this issue was opened, but now it's OK

**Comment by gisturbance** [comment]
> you have the option to search for specific track you want... ![image](https://private-user-images.githubusercontent.com/45605734/289645274-00d2707a-fbf5-4cc7-ab12-b6f0431048c8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI2MjMxOTMsIm5iZiI6MTcwMjYyMjg5MywicGF0aCI6Ii80NTYwNTczNC8yODk2NDUyNzQtMDBkMjcwN2EtZmJmNS00Y2M3LWFiMTItYjZmMDQzMTA0OGM4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE1VDA2NDgxM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ1ZTZkMjg4ODRiZmMxZTFkMTk0MTBjOGZjMmRlYjk3YTk4MTE1N2EwMTU2M2JjNTkxNTgzZGJhZmNmZjVmYjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.m-nFXZ0I54a1YmzEhsf7kGGaycRN7QIzzpEbKHgPJOk) pretty sure this wasn't available when this issue was opened, but now it's OK

This is useful when searching for a specific track, but not for downloading multiple individual tracks from an album found through album search.

**Comment by orbittwz** [comment]
this is correct yeah, but if you are downloading multiple tracks from the same album, why not keeping the whole album?

**Comment by gisturbance** [comment]
> this is correct yeah, but if you are downloading multiple tracks from the same album, why not keeping the whole album?

Possible reasons could be:
- you only like specific tracks of that album
- the album is a best-of collection with over 50 titles, you only want 17 specific titles
- you have a slow or limited connection - even if audio files are rather small

I would even go as far to suggest a unified search, that displays found tracks/albums together.

**Comment by orbittwz** [comment]
I did this and you could get specific tracks when you search anything...
mentioning this https://github.com/orbittwz/qbdLX/issues/16
do check it out!
https://github.com/orbittwz/qbdLX

**Comment by orbittwz** [comment]
definitely possible since I checked when testing 1.3.1...
select "Track" in the search, then write artist + album name, it will display all the tracks for said album.
enjoy and I'm marking this as solved with commit https://github.com/orbittwz/qbdLX/commit/629304217bf226ba30a887aecc605263e0b0f25a
